amethyst_derive/src/lib.rs

```diff
@@ -25,3 +26,10 @@ pub fn prefab_data_derive(input: TokenStream) -> TokenStream {
     let gen = prefab_data::impl_prefab_data(&ast);
     gen.into()
 }
+
+#[proc_macro_derive(WidgetId, attributes(widget_id))]
+pub fn widget_id_derive(input: TokenStream) -> TokenStream {
```

Comment:
are there docs around this macro?

Reply:
Not yet. My next PR will focus on providing detailed docs and examples on how to use Widgets (and probably even a small section in the book), as well as implementing them for every existing widget, so I wanted to put it off until then. If this has a high priority though, I can add a short description.

Reply:
Added a short description.